### DISCUSSION TITLE
Question about UCP_EP_PARAMS_FLAGS_CLIENT_SERVER flag

### DISCUSSION THREAD

**Discussion by huzhijiang** [openucx/ucx#9737] [General]
If I want to connect more than one ep between two nodes, should I make all connections using UCP_EP_PARAMS_FLAGS_CLIENT_SERVER ? Can I just make one connection using UCP_EP_PARAMS_FLAGS_CLIENT_SERVER (for wire up) while others using UCP_EP_PARAM_FIELD_REMOTE_ADDRESS method?

**Comment by yosefe** [comment]
> If I want to connect more than one ep between two nodes, should I make all connections using UCP_EP_PARAMS_FLAGS_CLIENT_SERVER ? Can I just make one connection using UCP_EP_PARAMS_FLAGS_CLIENT_SERVER (for wire up) while others using UCP_EP_PARAM_FIELD_REMOTE_ADDRESS method?

Yes, it is possible

**Reply by huzhijiang** [reply]
Thanks yosefe,
I would like to try. So The basic idea is make UCP_EP_PARAMS_FLAGS_CLIENT_SERVER ep connected first ,and then two nodes exchange other workers' addresses over that ep?

If above is right, what is the difference between the ep created with UCP_EP_PARAMS_FLAGS_CLIENT_SERVER flag and ep created using UCP_EP_PARAM_FIELD_REMOTE_ADDRESS method? I can treat them equaly? For example, Is the data sent/received through these two kinds of ep over the same set of iface resources?

**Reply by yosefe** [reply]
It doesn't matter in which order these endpoints are created , they are independent.
After the endpoints are connected, the data transfer works that same way on them, and they share the same iface resources.

**Comment by huzhijiang** [comment]
> It doesn't matter in which order these endpoints are created , they are independent. After the endpoints are connected, the data transfer works that same way on them, and they share the same iface resources.

Hi yosefe,

Got it. Thanks!
